### PR TITLE
fix(): missing requestBody

### LINKED ISSUE #1946: When using enableVersioning missing requestBody in document
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When using nestJs enableVersioning:

```
app.enableVersioning({
        defaultVersion: ['1', '2'],
        type: VersioningType.URI,
    });
```

And creating the document:

```
    const builder = new DocumentBuilder()
        .setTitle('Some API')
        .setVersion('1')
        .setDescription('This is a test swagger doc');
    const document = SwaggerModule.createDocument(app, builder.build(), {
      operationIdFactory: (controllerKey: string, methodKey: string) => methodKey,
    });

    // missing responseBody in each version
    console.log(JSON.stringify(document, null, 2));
```

requestBody is only available on the first version in the list /v1/example and NOT on /v2/example:

```
{
  "openapi": "3.0.0",
  "paths": {
    "/v1/example": {
      "post": {
        "operationId": "postHello",
        "summary": "",
        "description": "Some api operation to post hello",
        "parameters": [],
        "requestBody": {
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/HelloDto"
              }
            }
          }
        },
        "responses": {
          "201": {
            "description": ""
          }
        }
      }
    },
    "/v2/example": {
      "post": {
        "operationId": "postHello",
        "summary": "",
        "description": "Some api operation to post hello",
        "parameters": [],
        "responses": {
          "201": {
            "description": ""
          }
        }
      }
    }
  },
  "info": {
    "title": "Some API",
    "description": "This is a test swagger doc",
    "version": "1",
    "contact": {}
  },
  "tags": [],
  "servers": [],
  "components": {
    "schemas": {
      "HelloDto": {
        "type": "object",
        "properties": {
          "someProperty": {
            "type": "string",
            "description": "some property",
            "example": "pm_1DkAUlB70neqlGn3zQESm7pl"
          },
          "someOtherProperty": {
            "type": "string",
            "description": "some other property",
            "example": "pm_1DkAUlB70neqlGn3zQESm7pl"
          }
        }
      }
    }
  }
}

```

### Minimum reproduction code

https://github.com/getlabs-sean/nestjs-swaggerjs-enableVersioning-issue

### Steps to reproduce

1) `yarn install`
2) `yarn nest start`
3) see document dumped in console.log

### Expected behavior

We use code that separates the openapi.json file into versions and if the requestBody is not in each endpoint path we have broken documentation. 

### Package version

^5.2.1

### NestJS version

^8.0.0

### Node.js version

_No response_

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

_No response_

### PR DESCRIPTION
**Pull request by bhrigushr**
Closes #1946

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1946 

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by bhrigushr** [comment]
@micalevisk can please review this PR

**Comment by bhrigushr** [comment]
@micalevisk can you please review this PR

**Review by micalevisk — APPROVED** [review]

**Review by georgegrayos — APPROVED** [review]

File: test/explorer/swagger-explorer.spec.ts [hunk 1]
```diff
@@ -1349,4 +1349,79 @@ describe('SwaggerExplorer', () => {
       });
     });
   });
+
+  describe('when multiple version is defined', () => {
```
Comment (kamilmysliwiec) [inline test/explorer/swagger-explorer.spec.ts:1353]:
```suggestion
  describe('when multiple versions are defined', () => {
```

File: test/explorer/swagger-explorer.spec.ts [hunk 2]
```diff
@@ -1349,4 +1349,79 @@ describe('SwaggerExplorer', () => {
...
+          `/globalPrefix/v${
+            CONTROLLER_MULTIPLE_VERSIONS[0] as string
+          }/modulePath/with-multiple-version`
+        );
+        expect(routes[1].root.path).toEqual(
+          `/globalPrefix/modulePath/with-multiple-version`
+        );
+      });
+
+      it('should have requestBody in each versions of post route', () => {
```
Comment (kamilmysliwiec) [inline test/explorer/swagger-explorer.spec.ts:1408]:
```suggestion
      it('should have the requestBody in each version of POST route', () => {
```

File: test/explorer/swagger-explorer.spec.ts [hunk 3]
```diff
@@ -1349,4 +1349,79 @@ describe('SwaggerExplorer', () => {
...
+        explorer = new SwaggerExplorer(schemaObjectFactory);
+
+        config = new ApplicationConfig();
+        config.enableVersioning({
+          type: VersioningType.URI,
+          defaultVersion: VERSION_NEUTRAL
+        });
+      });
+
+      it('should use multiple versions defined', () => {
```
Comment (kamilmysliwiec) [inline test/explorer/swagger-explorer.spec.ts:1387]:
```suggestion
      it('should use multiple versions', () => {
```

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**